CMakeLists.txt

```diff
@@ -93,5 +93,5 @@ include(AzureVersion)
 add_subdirectory(sdk/core)
 add_subdirectory(sdk/identity)
 add_subdirectory(sdk/keyvault)
-add_subdirectory(sdk/storage)
+#add_subdirectory(sdk/storage)
```

Comment:
No. I have removed the fixes I did before for blobs as you requested.
Please fix and enable it back

Reply:
why it cannot be built? You only added something, you didn't remove anything or make any breaking changes.

Reply:
Yes, you are right. 
I have enabled it back.
I would like to remove the HttpPipeline constructor which takes the list of policies. That would be the breaking change

Reply:
> I would like to remove the HttpPipeline constructor which takes the list of policies. That would be the breaking change

You mean this one https://github.com/Azure/azure-sdk-for-cpp/blob/27943f0819e76eb74eaf8c884b5a93dad4f608f5/sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp#L162?

Reply:
Yes, that constructor is a way to workaround the mandated policies from Azure Core.
This was basically the arch board feedback and the reason of this entire PR.

The arch board is concerned about allowing services to choose all the policies for the pipeline, because they might decided not to add policies which Core wants to have in all SDK clients.

Reply:
Well, the arch board also mentioned that in the future, if it happen that we have 20 o 40 SDK services for C++, and we want to add a new client option to all of them, we want to do it only from Core and have all services updated.

Reply:
okay, i'm sorry for making you undo the changes to storage policy construction code. It's just I knew it wouldn't work when I saw your changes, mainly because auth policy has to be the last policy like I mentioned in another thread, and retry policy to secondary problem which we already have a solution. I just want to make sure everything, every details of this `ClientOptions` is thoroughly considered before we make changes to those stuff depend on it.